opengrok-indexer/src/test/java/org/opengrok/indexer/history/BoundaryChangesetsTest.java

```diff
@@ -74,6 +78,16 @@ void testInvalidMaxCount(int maxCount) {
         assertThrows(RuntimeException.class, () -> new BoundaryChangesets(gitSpyRepository));
     }
 
+    @Test
+    void testMaxCountConfiguration() {
+        int maxCount = 42 * new Random().nextInt(100) + 1;
+        assertNotEquals(0, maxCount);
+        RuntimeEnvironment.getInstance().setHistoryChunkCount(maxCount);
+        GitRepository gitSpyRepository = Mockito.spy(gitRepository);
```

Comment:
Why use of `spy` here? Won't it just delegate all the calls to `gitRepository`?

Reply:
good catch, fixed.